src/core/utils.js

```diff
@@ -822,3 +822,30 @@ export class InsensitiveStringSet extends Set {
         );
   }
 }
+
+export function makeSafeCopy(node) {
+  const clone = node.cloneNode(true);
+  clone.querySelectorAll("[id]").forEach(elem => elem.removeAttribute("id"));
+  clone.querySelectorAll("dfn").forEach(dfn => renameElement(dfn, "span"));
+  if (clone.hasAttribute("id")) clone.removeAttribute("id");
+  removeCommentNodes(clone);
+  return clone;
+}
+
+export function removeCommentNodes(node) {
+  const walker = document.createTreeWalker(node, NodeFilter.SHOW_COMMENT);
+  for (const comment of [...walkTree(walker)]) {
```

Comment:
Can we do this?
```suggestion
  for (const comment of walkTree(walker)) {
```

Reply:
I think I tried the previously and it breaks.

Reply:
let's try it and see what happens :)

Reply:
It won't work as it will cut the tree in the middle.

Reply:
yeah, I remember it did bad things... changing it back.